### PR TITLE
Dev/232 The ExamWhiteList component can't be rendered in the ExamOption component

### PR DESCRIPTION
**Pull request by Wally5077**
resolve: #232 

https://user-images.githubusercontent.com/26381900/140686743-ad666592-071c-463a-b2a1-3a4c6d0a7102.mov

### CODE REVIEW

File: src/app/exam/options/ExamWhiteList.js [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
-  const {textItems, addTextItem, removeTextItem} = useTextItems();
+const ExamWhiteList = ({whitelist, onWhitelistChange}) => {
+  const { textItems: ipAddresses, addTextItem: addIpAddress, removeTextItem: removeIpAddress } = useTextItems(whitelist);
+
+  useEffect(() => {
+    onWhitelistChange(ipAddresses);
+  }, [ipAddresses]);
+
+  const onAddIpAddress = (ipAddress) => {
+    const ipAddressFormat = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
```
Comment (Chaoyuuu) [inline src/app/exam/options/ExamWhiteList.js:16]: try this one `'\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.|$)){4}\b'`
here is the reference [link](https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp)

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review done

File: src/app/exam/options/ExamWhiteList.js (hunk 1 above)
Comment (Wally5077) [reply src/app/exam/options/ExamWhiteList.js:16]: Yes, it did work.

**Review by Chaoyuuu — APPROVED** [review]
LGTM

**Comment by snowmancc** [comment]
LGTM

**Review by snowmancc — APPROVED** [review]

**Pull request merged**